examples/hashing/src/main.sw

```diff
@@ -1,6 +1,73 @@
 script;
 
-use std::hash::{keccak256, sha256};
+use std::hash::*;
+
+impl Hash for str[4] {
```

Comment:
There's not much that can be done for this until we get const generics.
For strings in particular I guess the slice implementation should help.

Reply:
I am adding a generic implementation of the Hash trait that should cover any tuple with 2 to 5 elements and any array from 1 to 32 elements. We still have to write it once though on the hash.sw file, but it can cover any typed arrays and tuples, as long as the generic types implement Hash.